ext/openssl/openssl.c

```diff
@@ -7547,15 +7547,15 @@ static int php_openssl_cipher_init(const EVP_CIPHER *cipher_type,
 		return FAILURE;
 	}
 	if (mode->set_tag_length_always || (enc && mode->set_tag_length_when_encrypting)) {
-		if (!EVP_CIPHER_CTX_ctrl(cipher_ctx, mode->aead_set_tag_flag, tag_len, NULL)) {
+		if (EVP_CIPHER_CTX_ctrl(cipher_ctx, mode->aead_set_tag_flag, tag_len, NULL) != 1) {
```

Comment:
I just checked OpenSSL checks for `EVP_CTRL_AEAD_SET_TAG` and it usually uses ` <= 0` so think we should use that too. There was actually one usage `!= 1` so it might be safe but I would still prefer `<= 0`. Also this issue (checking for zero) is present in few places in OpenSSL itself...